pkg/controller/job/job_controller.go

```diff
@@ -579,39 +582,50 @@ func (jm *Controller) syncJob(key string) (bool, error) {
 			}
 		}
 		if complete {
-			job.Status.Conditions = append(job.Status.Conditions, newCondition(batch.JobComplete, v1.ConditionTrue, "", ""))
-			jobConditionsChanged = true
+			job.Status.Conditions, isUpdated = ensureJobConditionStatus(job.Status.Conditions, batch.JobComplete, v1.ConditionTrue, "Job completed", "Job completed")
+			jobConditionsChanged = jobConditionsChanged || isUpdated
 			now := metav1.Now()
 			job.Status.CompletionTime = &now
 			jm.recorder.Event(&job, v1.EventTypeNormal, "Completed", "Job completed")
-		} else if utilfeature.DefaultFeatureGate.Enabled(features.SuspendJob) && manageJobCalled {
-			// Update the conditions / emit events only if manageJob was called in
-			// this syncJob. Otherwise wait for the right syncJob call to make
-			// updates.
-			if job.Spec.Suspend != nil && *job.Spec.Suspend {
-				// Job can be in the suspended state only if it is NOT completed.
-				var isUpdated bool
-				job.Status.Conditions, isUpdated = ensureJobConditionStatus(job.Status.Conditions, batch.JobSuspended, v1.ConditionTrue, "JobSuspended", "Job suspended")
-				if isUpdated {
-					jobConditionsChanged = true
-					jm.recorder.Event(&job, v1.EventTypeNormal, "Suspended", "Job suspended")
-				}
-			} else {
-				// Job not suspended.
-				var isUpdated bool
-				job.Status.Conditions, isUpdated = ensureJobConditionStatus(job.Status.Conditions, batch.JobSuspended, v1.ConditionFalse, "JobResumed", "Job resumed")
-				if isUpdated {
-					jobConditionsChanged = true
-					jm.recorder.Event(&job, v1.EventTypeNormal, "Resumed", "Job resumed")
-					// Resumed jobs will always reset StartTime to current time. This is
-					// done because the ActiveDeadlineSeconds timer shouldn't go off
-					// whilst the Job is still suspended and resetting StartTime is
-					// consistent with resuming a Job created in the suspended state.
-					// (ActiveDeadlineSeconds is interpreted as the number of seconds a
-					// Job is continuously active.)
-					now := metav1.Now()
-					job.Status.StartTime = &now
-				}
+		} else {
+			job.Status.Conditions, isUpdated = ensureJobConditionStatus(job.Status.Conditions, batch.JobComplete, v1.ConditionFalse, "", "")
+			jobConditionsChanged = jobConditionsChanged || isUpdated
+		}
+
+		if !utilfeature.DefaultFeatureGate.Enabled(features.SuspendJob) {
```

Comment:
Consider putting all this logic in a function.

For example, you could just return after removing the condition if the gate is disabled, simplifying the conditionals below.